From d60fcd5bd5c2675e4342775b910a2ea48ec0eccb Mon Sep 17 00:00:00 2001
From: Dimitry Andric <dim@FreeBSD.org>
Date: Wed, 19 Aug 2020 03:35:16 +0000
Subject: [PATCH] scanner: use macro instead of cast to convert pointer to integer

Clang 11 build failed due to a new warning (part of -Werror=pointer-to-int-cast):
../json-glib/json-scanner.c:928:13: error: cast to smaller integer type 'GTokenType' from 'gpointer' (aka 'void *') [-Werror,-Wvoid-pointer-to-enum-cast]
        *token_p = (GTokenType) value_p->v_symbol;
                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Upstream-Status: Backport [https://gitlab.gnome.org/GNOME/json-glib/-/commit/8c5fabe962b7337066dac7a697d23fce257a5d64]
Signed-off-by: Jan Beich <jbeich@FreeBSD.org>
Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 json-glib/json-scanner.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/json-glib/json-scanner.c b/json-glib/json-scanner.c
index 0c9919f..59dd29c 100644
--- a/json-glib/json-scanner.c
+++ b/json-glib/json-scanner.c
@@ -925,7 +925,7 @@ json_scanner_get_token_i (JsonScanner	*scanner,
       
     case G_TOKEN_SYMBOL:
       if (scanner->config->symbol_2_token)
-	*token_p = (GTokenType) value_p->v_symbol;
+	*token_p = GPOINTER_TO_INT (value_p->v_symbol);
       break;
       
     case G_TOKEN_BINARY:
-- 
2.28.0

